signatory-sgx/src/provider.rs

```diff
@@ -0,0 +1,89 @@
+use crate::error::Error;
```

Comment:
perhaps there should be some SgxSigner struct in this provider that will implement the actual traits like `signatory:: public_key::PublicKeyed<signatory::ed25519::PublicKey>` and `signatory::signature::Signer<signatory::ed25519::Signature>`?